python_tests/pytest_input_generators.py

```diff
@@ -360,6 +360,59 @@ def define_vector_input_error_generator(
         yield SampleInput(**es.kwargs), es.ex_type, es.ex_str
 
 
+# TODO Add small value, large value, and extremal-valued samples
+def elementwise_binary_generator(
+    op: OpInfo,
+    dtype: torch.dtype,
+    requires_grad: bool = False,
+    *,
+    supports_numbers: bool = True,
+    **kwargs,
+):
+    low = None if op.domain.low is None else max(-9, op.domain.low)
+    high = None if op.domain.high is None else min(9, op.domain.high)
+    make_arg = partial(
+        make_tensor,
+        device="cuda",
+        dtype=dtype,
+        low=low,
+        high=high,
+        requires_grad=requires_grad,
+        **kwargs,
+    )
+
+    shapes = (
+        # TODO: restore size zero cases
```

Comment:
is size zero broken? do we have an issue tracking it?

Reply:
Size zero works fine. I took these test cases from elsewhere.